doc/source/basics.rst

```diff
@@ -1159,14 +1159,17 @@ mapping (a dict or Series) or an arbitrary function.
    s.rename(str.upper)
 
 If you pass a function, it must return a value when called with any of the
-labels (and must produce a set of unique values). But if you pass a dict or
-Series, it need only contain a subset of the labels as keys:
+labels (and must produce a set of unique values). A dict or
+Series can also be used:
 
 .. ipython:: python
 
    df.rename(columns={'one' : 'foo', 'two' : 'bar'},
              index={'a' : 'apple', 'b' : 'banana', 'd' : 'durian'})
 
+If mapping doesn't name a column/index, it isn't renamed. Also extra names
```

Comment:
"If mapping doesn't name a column/index," -> "If the mapping doesn't include a column/index label"